gluten-ut/spark40/src/test/scala/org/apache/spark/sql/connector/GlutenKeyGroupedPartitioningSuite.scala

```diff
@@ -72,10 +84,46 @@ class GlutenKeyGroupedPartitioningSuite
       case s: SortMergeJoinExec => s
     }.flatMap(smj => collect(smj) { case s: ColumnarShuffleExchangeExec => s })
   }
+
+  private def collectShuffles(plan: SparkPlan): Seq[ShuffleExchangeLike] = {
```

Comment:
Will create a PR in apache/Spark to make the function from private to protected, then we can only override the function to check the plan